ktor-shared/ktor-serialization/ktor-serialization-gson/jvm/src/GsonConverter.kt

```diff
@@ -59,6 +84,33 @@ public class GsonConverter(private val gson: Gson = Gson()) : ContentConverter {
             throw JsonConvertException("Illegal json parameter found", deserializeFailure)
         }
     }
+
+    private val prettyPrinting by lazy {
+        val field = Gson::class.java.getDeclaredField("prettyPrinting")
+        field.isAccessible = true
+        field.get(gson) as Boolean
+    }
+
+    private companion object {
+        private const val beginArrayCharCode = '['.code
+        private const val endArrayCharCode = ']'.code
+        private const val objectSeparator = ','.code
+    }
+
+    /**
+     * Guaranteed to be called inside a [Dispatchers.IO] context, see [OutputStreamContent]
+     */
+    @Suppress("BlockingMethodInNonBlockingContext")
+    private suspend fun <T> Flow<T>.serializeJson(writer: Writer) {
+        writer.write(beginArrayCharCode)
+        collectIndexed { index, value ->
+            if (index > 0) {
+                writer.write(objectSeparator)
+            }
+            gson.toJson(value, writer)
```

Comment:
what do you think about flushing after you wrote value and separator, so that the receiver side can start processing the item?